app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -49,6 +52,7 @@ class SubmissionFileManager extends React.Component {
   // Update state when a new revision_identifier props is passed
   componentDidUpdate(oldProps) {
     if (oldProps.revision_identifier !== this.props.revision_identifier) {
+      this.setState({viewFile: null, viewFileType: null});
```

Comment:
Move this `setState` call into the `fetchData`'s "then". (Again, merging the `setState` calls.)